lib/plugins/aws/deploy/compile/events/apiGateway/lib/methods.js

```diff
@@ -5,13 +5,27 @@ const _ = require('lodash');
 
 module.exports = {
   compileMethods() {
+    const corsConfig = {};
     let endpointCounter = 1;
-
     _.forEach(this.serverless.service.functions, (functionObject, functionName) => {
       functionObject.events.forEach(event => {
         if (event.http) {
           let method;
           let path;
+          let corsEnabled = false;
+
+          const headers = [
```

Comment:
Can you move this `const headers` and the `let cors = {` below into the `if (!!event.http.cors)` section below so that the variables are only created when CORS is enabled?